tests/features/editorial_workflow.feature

```diff
@@ -0,0 +1,140 @@
+@api
+Feature: Corporate editorial workflow
+  As a content editor
+  I can moderate content through different states by different roles
+
+  Scenario: As an Author user, I can create a Draft demo content and I can send it for Needs Review
+    Given I am logged in as a user with the "Author" role
+    When I visit "node/add/oe_workflow_demo"
+    Then I should have the following options for "Save as" select:
+      | Draft        |
+      | Needs Review |
+    When I fill in "Title" with "Workflow demo"
+    And I press "Save"
+    And I visit "admin/content"
+    # The content is created and it is not published.
+    Then I should see text matching "Workflow demo"
```

Comment:
Even if in this case you only have one node, this should be a step that check that on that specific table row the status is "not published". You can use the `DrupalContext::assertTextInTableRow`